Tests/AsyncQueueTests/FIFOQueueTests.swift

```diff
@@ -65,7 +91,26 @@ final class FIFOQueueTests: XCTestCase {
         await systemUnderTest.await { /* Drain the queue */ }
     }
 
-    func test_async_isNotReentrant() async {
```

Comment:
this diff is slightly confusingly presented: I renamed this test to be more clear, and added a new test above this one.